File: backend/app/schemas/document_classifier.py
"""Schemas for the hybrid document classification pipeline."""

from pydantic import BaseModel, ConfigDict


class ClassificationCandidate(BaseModel):
    """A candidate document type with confidence score."""

    type: str
    confidence: float

    model_config = ConfigDict(from_attributes=True)


class ClassificationResult(BaseModel):
    """Result from classifying a single document."""

    document_id: str | None = None
    document_type: str
    confidence: float
    method: str  # filename | content | hybrid
    candidates: list[ClassificationCandidate]
    ai_generated: bool = True
    keywords_found: list[str]

    model_config = ConfigDict(from_attributes=True)


class BatchClassificationResult(BaseModel):
    """Summary of batch classification for a building."""

    building_id: str
    total_processed: int
    classified_count: int
    unclassified_count: int
    results: list[ClassificationResult]

    model_config = ConfigDict(from_attributes=True)


class DocumentTypeInfo(BaseModel):
    """Info about a supported document type."""

    type_key: str
    label_fr: str
    label_en: str
    label_de: str
    label_it: str
    keywords: list[str]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/documentClassifier.ts
import { apiClient } from '@/api/client';

export interface ClassificationCandidate {
  type: string;
  confidence: number;
}

export interface ClassificationResult {
  document_id: string | null;
  document_type: string;
  confidence: number;
  method: 'filename' | 'content' | 'hybrid';
  candidates: ClassificationCandidate[];
  ai_generated: boolean;
  keywords_found: string[];
}

export interface BatchClassificationResult {
  building_id: string;
  total_processed: number;
  classified_count: number;
  unclassified_count: number;
  results: ClassificationResult[];
}

export interface DocumentTypeInfo {
  type_key: string;
  label_fr: string;
  label_en: string;
  label_de: string;
  label_it: string;
  keywords: string[];
}

export const documentClassifierApi = {
  classifySingle: async (documentId: string): Promise<ClassificationResult> => {
    const response = await apiClient.post<ClassificationResult>(`/documents/${documentId}/classify`);
    return response.data;
  },

  batchClassify: async (buildingId: string): Promise<BatchClassificationResult> => {
    const response = await apiClient.post<BatchClassificationResult>(
      `/buildings/${buildingId}/documents/classify-all`,
    );
    return response.data;
  },

  listTypes: async (): Promise<DocumentTypeInfo[]> => {
    const response = await apiClient.get<DocumentTypeInfo[]>('/documents/types');
    return response.data;
  },
};
